src/System.Net.Requests/tests/RequestStreamTest.cs

```diff
@@ -11,11 +11,18 @@
 
 namespace System.Net.Tests
 {
-    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Tests hang on .NET Framework")]
     public class RequestStreamTest
     {
         readonly byte[] buffer = new byte[1];
 
+        static RequestStreamTest()
+        {
+            // On .NET Framework, the default limit for connections/server is very low (2).
+            // On .NET Core, the default limit is higher. Since these tests run in parallel,
+            // the limit needs to be increased to avoid timeouts when running the tests.
+            ServicePointManager.DefaultConnectionLimit = int.MaxValue;
```

Comment:
I want to clarify that this is not a bug in Desktop.  This is part of the design of HttpWebRequest and GetRequestStream.

It should not close the TCP connection when you close/dispose/using the request stream.

The reason is that the semantics are that you are supposed to get the request stream, write to it, and then close/dispose it.  That completes the "request" portion of the HTTP request.  For cases where AllowWriteStreamBuffering is true, it will calculate the content size and adjust the Content-Length header.  For cases of chunked transfer, it will send the zero-length terminator.

But it should not close the TCP connection.  The connection has to stay open in order to be ready for someone to ask for the response via GetResponse.